src/xrc/src/api/test.rs

```diff
@@ -117,13 +111,15 @@ impl TestCallExchangesImplBuilder {
 impl CallExchanges for TestCallExchangesImpl {
     async fn get_cryptocurrency_usdt_rate(
         &self,
+        exchanges: &[&Exchange],
         asset: &Asset,
         timestamp: u64,
-    ) -> Result<QueriedExchangeRate, CallExchangeError> {
+    ) -> Result<QueriedExchangeRateWithFailedExchanges, CallExchangeError> {
+        let exchanges_vec = exchanges.iter().map(|e| e.to_owned().clone()).collect::<Vec<_>>();
```

Comment:
Isn't there an easier way to accomplish this?
Why doesn't exchanges.clone() work?

Reply:
`get_stablecoin_rates_calls` expects `Vec<Exchange>`, and when you do `exchanges.clone()` the type is `&[&Exchange]`

Reply:
What about `exchanges.iter().cloned()`?   :slightly_smiling_face: